### ISSUE TITLE
Compressed string type?

### ISSUE THREAD

**Issue by jenstroeger** [kvesteri/sqlalchemy-utils#380]
Would it make sense and is there interest in a [data type](https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html) for compressed unicode strings? Python has builtin support for compression (e.g. [bz2](https://docs.python.org/3/library/bz2.html) or [lzma](https://docs.python.org/3/library/lzma.html)) so adding a type that receives `str` and compresses them for blob columns would in some cases be useful.

**Comment by olirice** [comment]
It would be straightforward to implement, but it introduces some risks:

Assuming something like this:

```python
import sqlalchemy as sa
import gzip

class GZIPString(types.TypeDecorator):
    impl = sa.types.LargeBinary

    def process_bind_param(self, value, dialect):
        return gzip.compress(bytes(exampleString,'utf-8'))

    def process_result_value(self, value, dialect):
        return gzip.decompress(compressed_value)
```

It would compress on save and decompress on read.

If the data being stored is well behaved, the risks/downsides are:
- it'd use a lot of cpu on insert/update/query
- it would consume an unknown amount of memory when querying

If the data being stored is potentially malicious (user defined input) it wouldn't be hard to use those downsides to overload your server

**Comment by jenstroeger** [comment]
@olirice, yes I agree with everything you mention (implementation and risks) and I would assume that users of such a type would know what they’re doing. For my particular use case, the string data would be very controlled (XML files) and read/write performance is not critical.
